modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManagerTest.java

```diff
@@ -0,0 +1,881 @@
+
+        Group groupOld = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING, schema, new Group());
+        String name = groupOld.getId();
+
+        return new Object[][]{
+                {name, NEW_GROUP_SCIM_OBJECT_STRING_UPDATED, "id", null, groupOld}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestUpdateGroupWithPUTUpdatedGroupResourceIsNull")
+    public void testUpdateGroupWithPUTUpdatedGroupResourceIsNull(String id, String scimObjectString, String
+            attributes, String excludeAttributes, Object objectOLDGroup)
+            throws BadRequestException, NotFoundException, CharonException, NotImplementedException {
+
+        Group groupOld = (Group) objectOLDGroup;
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT + "/" + GROUP_ID);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(InternalErrorException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new InternalErrorException()));
+
+        when(userManager.getGroup(id,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenReturn(groupOld);
+
+        when(userManager.updateGroup(anyObject(), anyObject(), anyObject())).thenReturn(null);
+
+        SCIMResponse scimResponse = groupResourceManager.updateWithPUT(id, scimObjectString, userManager,
+                attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_INTERNAL_ERROR);
+    }
+
+    @DataProvider(name = "dataForTestUpdateGroupWithPUTNotFoundException")
+    public Object[][] dataToTestUpdateGroupWithPUTNotFoundException()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        Group groupOld = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING, schema, new Group());
+        String id = groupOld.getId();
+
+        Group groupNew = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING_UPDATED, schema, new Group());
+
+        return new Object[][]{
+                {id, NEW_GROUP_SCIM_OBJECT_STRING_UPDATED, "id", null, groupNew, groupOld}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestUpdateGroupWithPUTNotFoundException")
+    public void testUpdateGroupWithPUTNoUserExistsWithTheGivenUserName(String id, String scimObjectString, String
+            attributes, String excludeAttributes, Object objectNEWGroup, Object objectOLDGroup)
```

Comment:
`objectNEWGroup` ->  As a convention, if variable names consist of multiple words, each word after the first word has its first letter written in uppercase and followed by the rest of the letters in the word in lowercase.

therefore I suggest having it as something like `newGroup`. Check all applicable places.

Reply:
Followed the camel case convention and fixed in commit 1c9cf8aca69b6c33157e750da50f23315f9947fc